services/src/main/java/org/apache/druid/cli/CliCoordinator.java

```diff
@@ -353,4 +370,76 @@ public static boolean isOverlord(Properties properties)
   {
     return Boolean.parseBoolean(properties.getProperty(AS_OVERLORD_PROPERTY));
   }
+
+  private static class CoordinatorCustomDutyGroupsProvider implements Provider<CoordinatorCustomDutyGroups>
+  {
+    private Properties props;
+    private JsonConfigurator configurator;
+    private ObjectMapper jsonMapper;
+
+    @Inject
+    public void inject(Properties props, JsonConfigurator configurator, ObjectMapper jsonMapper)
+    {
+      this.props = props;
+      this.configurator = configurator;
+      this.jsonMapper = jsonMapper;
+    }
+
+    @Override
+    public CoordinatorCustomDutyGroups get()
```

Comment:
is there a test for this deserialization?

Reply:
There is no unit test since it is difficult to setup all the modules/injectors. However, this is test in the Integration test. The integration tests (`common-custom-coordinator-duties`) set the following config: 
```
druid.coordinator.dutyGroups=["cleanupMetadata"]
druid.coordinator.cleanupMetadata.duties=["killSupervisors"]
druid.coordinator.cleanupMetadata.duty.killSupervisors.retainDuration=PT0M
druid.coordinator.cleanupMetadata.period=PT10S
``` 
and verify that the custom "killSupervisors" is created correctly and does its intended duty successfully.